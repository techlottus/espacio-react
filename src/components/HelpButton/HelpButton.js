import React from "react";
import "./HelpButton.scss";
import { Label } from "../Label/Label";
import { ButtonCircle } from "../ButtonCircle/ButtonCircle";
import {
  btnCircleHelpCenterDashboard,
  labelHelpCenterDashboard,
} from "../../constants/DashboardHelpCenter.constant";

const HelpButton = ({name}) => {

  return (
    <div className="helpbutton-container">
      <div className="helpbutton-content">
        <Label data={labelHelpCenterDashboard} prop={name}/>
      </div>
      <ButtonCircle data={btnCircleHelpCenterDashboard} />
    </div>
  );
};

export default HelpButton;
