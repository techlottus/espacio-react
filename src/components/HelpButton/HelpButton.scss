@import "../../App.scss";

.helpbutton-container {
  display: flex;
  align-items: center;

  .helpbutton-content {
    margin-right: pixeltorem(16px);
  }
}
