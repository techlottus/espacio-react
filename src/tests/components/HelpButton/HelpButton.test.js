import React from "react";
import { mount } from "enzyme";
import { Provider } from "react-redux";
import "@testing-library/jest-dom";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";
import HelpButton from "../../../components/HelpButton/HelpButton";

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

let initState;
let name = "";

let store = mockStore(initState);
const handleMount = (initState) => {
  store = mockStore(initState);
  return mount(
    <Provider store={store}>
      <HelpButton name={name} />
    </Provider>
  );
};

describe("Test <HelpButton/>", () => {
  test("should init component", () => {
    const wrapper = handleMount(initState);
    expect(wrapper).toMatchSnapshot();
    wrapper.unmount();
  });
});
