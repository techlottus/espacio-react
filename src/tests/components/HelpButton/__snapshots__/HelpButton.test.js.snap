// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <HelpButton/> should init component 1`] = `
<div
  className="helpbutton-container"
>
  <div
    className="helpbutton-content"
  >
    <lottus-label />
  </div>
  <lottus-button-circle />
</div>
`;
